src/app/app-interactive-map/app-interactive-map.component.ts

```diff
@@ -127,23 +133,79 @@ export class AppInteractiveMapComponent implements OnInit, AfterViewInit {
+    }));
   }
 
-  handleSetAsObjective(args: string): void {
-    console.log('SET AS OBJECTIVE', args);
+  handleSetAsObjective(reactionId: string): void {
+    this.store.dispatch(new fromActions.SetObjectiveReaction({
+      reactionId,
+      direction: 'max',
+    }));
   }
 
-  handleChangeBounds(args: string): void {
-    console.log('CHANGE BOUNDS', args);
+  handleChangeBounds(reactionId: string): void {
+    this.store.dispatch(new fromActions.ReactionOperation({
+      item: {
+        reaction: this.findReaction(this.card.model.reactions, reactionId),
+        lowerBound: parseInt(this.elRef.nativeElement.querySelector('#lowerbound').value, 10),
+        upperBound: parseInt(this.elRef.nativeElement.querySelector('#upperbound').value, 10),
+      },
+      operationTarget: 'bounds',
+      direction: OperationDirection.Do,
+    }));
   }
 
-  handleResetBounds(args: string): void {
-    console.log('RESET BOUNDS', args);
+  handleResetBounds(reactionId: string): void {
+    this.store.dispatch(new fromActions.ReactionOperation({
+      item: <Bound>(this.findBoundReaction(reactionId)),
+      operationTarget: 'bounds',
+      direction: OperationDirection.Undo,
+    }));
   }
 
-  handleObjectiveDirection(args: string): void {
-    console.log('OBJECTIVE DIRECTION', args);
+  handleObjectiveDirection(reactionId: string): void {
+    this.store.dispatch(new fromActions.SetObjectiveReaction({
+      reactionId,
+      direction: this.card.objectiveReaction.direction === 'max' ? 'min' : 'max',
+    }));
+  }
+
+  checkStateObjective(reactionId: string): boolean {
+    if (this.card.objectiveReaction) {
+      return (this.card.objectiveReaction.reactionId === reactionId && this.card.objectiveReaction.direction === 'max');
+    } else {
+      return false;
+    }
+  }
+
+  upperBound(reactionId: string): number {
+    const reaction = this.findBoundReaction(reactionId);
+    return (<Cobra.Reaction>reaction).upper_bound ? (<Cobra.Reaction>reaction).upper_bound : (<Bound>reaction).upperBound;
+  }
+
+  lowerBound(reactionId: string): number {
+    const reaction = this.findBoundReaction(reactionId);
+    return (<Cobra.Reaction>reaction).lower_bound ? (<Cobra.Reaction>reaction).upper_bound : (<Bound>reaction).lowerBound;
+  }
+
+  knockoutText(reactionId: string): string {
+    return this.card.knockoutReactions && this.card.knockoutReactions.includes(reactionId) ? 'Undo knockout' : 'Knockout';
+  }
+
+  objectiveText(reactionId: string): string {
+    return this.card.objectiveReaction && this.card.objectiveReaction.reactionId === reactionId ? 'Undo set as objective' : 'Set as objective';
+  }
+
+  findBoundReaction (reactionId: string): Bound | Cobra.Reaction {
+    return this.card.bounds.find((r) => r.reaction.id === reactionId) || this.findReaction(this.card.model.reactions, reactionId);
+  }
+
+  findReaction (reactions: Cobra.Reaction[], reactionId: string): Cobra.Reaction {
```

Comment:
I think this utility method is a bit overkill, it's straight forward to do `...reactions.find((r) => r.id === reactionId)` where needed.

Reply:
I agree.